tests/DataTransferObjectTest.php

```diff
@@ -501,4 +501,33 @@ public function ignore_static_public_properties_when_serializing()
 
         $this->assertSame($expected, $object->all());
     }
+
+    /** @test */
+    public function multiple_required_properties_are_reported()
+    {
+        $exception = null;
+
+        try {
+            new class() extends DataTransferObject {
+                public string $i_am_required;
+                public int $so_am_i;
+            };
+        } catch (DataTransferObjectError $exception) {
+            // Deliberately empty
+        }
+
+        $this->assertInstanceOf(DataTransferObjectError::class, $exception);
+        
+        $actual = $exception->getMessage();
+        $actual = preg_replace('/anonymous.*:/', 'anonymous:', $actual);
```

Comment:
I decided not to use `expectExceptionMessageRegExp` because the regex for the muliltine message was crazy to formulate and hardly readable and I figured: clarity trumps so I just catch it => massage the message to remove the environment dependent paths => then simply match it 1:1